### PR TITLE
CDAP-18262 do not finish computing summary if it is stuck for too long

### PR DESCRIPTION
**Pull request by yaojiefeng**
The library we use will take > 5s for a value of string > 1000 characters, add this workaround to timeout the computation.

### CODE REVIEW

File: wrangler-core/src/main/java/io/cdap/wrangler/statistics/BasicStatistics.java [hunk 1]
```diff
@@ -55,10 +58,15 @@ public Row aggregate(List<Row> rows) {
           if (value.isEmpty()) {
             stats.increment(column, "empty");
           } else {
+            // this call is very expensive for string > 2000 characters, took seconds to return
             Map<String, List<String>> finds = engine.findWithType(value);
             for (String find : finds.keySet()) {
               types.increment(column, find);
             }
+            // TODO: this is a workaround for CDAP-18262, to proper fix we should revisit this computation logic
```
Comment (seanzhougoogle) [inline wrangler-core/src/main/java/io/cdap/wrangler/statistics/BasicStatistics.java:66]: file a TODO jira for this ? or just leave CDAP-18262 open and mark fix version as 6.6.0 ?

**Review by seanzhougoogle — APPROVED** [review]

File: wrangler-core/src/main/java/io/cdap/wrangler/statistics/BasicStatistics.java (hunk 1 above)
Comment (yaojiefeng) [reply wrangler-core/src/main/java/io/cdap/wrangler/statistics/BasicStatistics.java:66]: I will file a new JIRA for this and close 18262

File: wrangler-core/src/main/java/io/cdap/wrangler/statistics/BasicStatistics.java [hunk 2]
```diff
@@ -55,10 +58,15 @@ public Row aggregate(List<Row> rows) {
...
+            if (System.currentTimeMillis() - now > TIME_OUT_MILLIS) {
```
Comment (albertshau) [inline wrangler-core/src/main/java/io/cdap/wrangler/statistics/BasicStatistics.java]: now is a confusing variable name, better to name it start or something like that

File: wrangler-core/src/main/java/io/cdap/wrangler/statistics/BasicStatistics.java (hunk 1 above)
Comment (yaojiefeng) [reply wrangler-core/src/main/java/io/cdap/wrangler/statistics/BasicStatistics.java:66]: https://cdap.atlassian.net/browse/CDAP-18277

File: wrangler-core/src/main/java/io/cdap/wrangler/statistics/BasicStatistics.java (hunk 2 above)
Comment (yaojiefeng) [reply wrangler-core/src/main/java/io/cdap/wrangler/statistics/BasicStatistics.java]: Fixed

**Pull request merged**